aspnetcore/fundamentals/startup.md

```diff
@@ -1,75 +1,117 @@
+* Can optionally include a [ConfigureServices](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configureservices) method to configure the app's services.
+* Must include a [Configure](/dotnet/api/microsoft.aspnetcore.hosting.startupbase.configure) method to create the app's request processing pipeline.
 
-Alternatively, you can define a fixed `Startup` class that will be used regardless of the environment by calling `UseStartup<TStartup>`. This is the recommended approach.
+`ConfigureServices` and `Configure` are called by the runtime when the app starts:
 
-The `Startup` class constructor can accept dependencies that are provided through [dependency injection](xref:fundamentals/dependency-injection). A common approach is to use `IHostingEnvironment` to set up [configuration](xref:fundamentals/configuration/index) sources.
+[!code-csharp[Main](startup/snapshot_sample/Startup1.cs)]
 
-The `Startup` class must include a `Configure` method and can optionally include a `ConfigureServices` method, both of which are called when the application starts. The class can also include [environment-specific versions of these methods](xref:fundamentals/environments#startup-conventions). `ConfigureServices`, if present, is called before `Configure`.
+Specify the startup class with the [WebHostBuilderExtensions](/dotnet/api/Microsoft.AspNetCore.Hosting.WebHostBuilderExtensions) [UseStartup&lt;TStartup&gt;](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderextensions.usestartup#Microsoft_AspNetCore_Hosting_WebHostBuilderExtensions_UseStartup__1_Microsoft_AspNetCore_Hosting_IWebHostBuilder_) method:
 
-Learn about [handling exceptions during application startup](xref:fundamentals/error-handling#startup-exception-handling).
+[!code-csharp[Main](../common/samples/WebApplication1DotNetCore2.0App/Program.cs?name=snippet_Main&highlight=10)]
+
+The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment`), and the appropriate startup class is selected at runtime by convention. The class whose name suffix matches the current environment is prioritized. If the app is run in the Development environment and includes both a `Startup` class and a `StartupDevelopment` class, the `StartupDevelopment` class is used. See [Working with multiple environments](xref:fundamentals/environments#startup-conventions) for more information.
+
+The `Startup` class constructor can accept dependencies defined by the host. A common use of DI with the `Startup` class is to inject [IHostingEnvironment](/dotnet/api/Microsoft.AspNetCore.Hosting.IHostingEnvironment). This approach can be used to configure services as an alternative to the convention-based approach described above:
```

Comment:
~can~ accepts

Reply:
I think I'd rewrite this section something like:

The ASP.NET Core templates use DI to inject dependencies defined by the host with `IHostingEnvironment`.   or something like that.

Reply:
I re-worked this section a bit, but I still need to say that `IHostingEnvironment` is a common way to configure services by injecting it into `Startup` like this: (example). I move this up in the coverage before the conventions-based approach.